### PR TITLE
feat(weekly.ci) PVC for weekly.ci.jio

### PR DESCRIPTION
**Pull request by smerle33**
as per https://github.com/jenkins-infra/helpdesk/issues/4044

create a dedicated PVC for weekly.ci that will create a PV and Disk

### CODE REVIEW

**Comment by smerle33** [comment]
Plan: 1 to add, 0 to change, 0 to destroy.

File: weekly.ci.jenkins.io.tf [hunk 1]
```diff
@@ -0,0 +1,15 @@
+resource "kubernetes_persistent_volume_claim" "jenkins_weekly_PVC" {
+  metadata {
+    name = "jenkins-weekly-pvc"
```
Comment (dduportal) [inline weekly.ci.jenkins.io.tf:3]:
```suggestion
resource "kubernetes_persistent_volume_claim" "jenkins_weekly_data" {
  metadata {
    name = "jenkins-weekly-data"
```

File: weekly.ci.jenkins.io.tf [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+resource "kubernetes_persistent_volume" "jenkins_weekly_data" {
+  provider               = kubernetes.publick8s
+  metadata {
+    name = "jenkins-weekly-pv"
+  }
+  spec {
+    capacity = {
+      storage = local.weekly_ci_disk_size
+    }
+    access_modes                     = ["ReadWriteOnce"]
```
Comment (dduportal) [inline weekly.ci.jenkins.io.tf:27]: Nitpick: factorize

Comment (smerle33) [reply weekly.ci.jenkins.io.tf]: Plan: 4 to add, 0 to change, 0 to destroy.

File: locals.tf [hunk 3]
```diff
@@ -51,4 +51,7 @@ locals {
 
   publick8s_compute_zones            = [3]
   cijenkinsio_agents_1_compute_zones = [1]
+
+  weekly_ci_disk_size    = 8
+  weekly_ci_access_modes = "ReadWriteOnce"
```
Comment (dduportal) [inline locals.tf:56]:
```suggestion
  weekly_ci_access_modes = ["ReadWriteOnce"]
```

you want to factorize the type "list of string"

**Review by dduportal — APPROVED** [review]

**Comment by smerle33** [comment]
with both CSI and a statically provisioned storage class it worked : 

Apply complete! Resources: 4 added, 0 changed, 0 destroyed

**Pull request merged**